### ISSUE TITLE
Accessibility issues

### ISSUE THREAD

**Issue by james-pearce-bath-ac** [studentquiz/moodle-mod_studentquiz#482]
We have had one of our team review the plugin in terms of accessibility, he has observed the following issues. I'll add them in total here but happy to break these into individual issues if easier to complete.

### Attempting a quiz
**Issue:**

- Fast filters ARIA - for screenreaders the status of these filters is not conveyed to screenreaders (i.e. if there is no ARIA telling a screenreader if "Unanswered" is selected or not). MAJOR
- If there is an existing personal rating, this is not clearly read out MAJOR
- Related: confusing - if student presses Next on a question without rating, this shifts focus back to the Rate field (good), but it's easy to miss that this is a required field (and nothing is selected).
- Autotesting flags missing role attribute on the <span>
- The progress on the grey/blue "n questions bar" is not communicated to screenreaders (this visual shading is for sighted users only). (it should have text "n/N questions completed")

 
**UX**

- Add public comment is after the Next/Finish button - it is easy to miss this out on a screen reader
- Change state from New dropdown- ARIA text is not really descriptive enough to understand the context.
- There is no quiz navigation block in the block drawer

 
### Main activity page
**Issue:**

- Question state in table (new[default]/reviewable/changed)
- it is not clear what these icons mean to sighted users.
- These are not read aloud to screenreaders at all

 
**UX**

- Under My Attempts/Latest, I don't understand what the latest column is for. Ideally it shouldn't be a character tick/cross but seems to be OK with screenreader.

 
### Block drawer - My Progress
**UX**

- latest attempt and questions never answered progress bar aren't conveyed to screenreaders
